data_model/src/block.rs

```diff
@@ -175,12 +220,26 @@ mod committed {
     }
 
     impl CommittedBlock {
-        /// Calculate the hash of the current block.
-        /// `CommitedBlock` should have the same hash as `ValidBlock`.
+        /// Calculate the partial hash of the current block.
+        /// [`CommitedBlock`] should have the same partial hash as [`PendingBlock`].
+        #[cfg(feature = "std")]
+        #[inline]
+        pub fn partial_hash(&self) -> PartialBlockHash {
```

Comment:
this is only used by `iroha_core`. If it will never be used by `iroha_client` or `iroha_wasm` then you should move this into `iroha_core`. I know this is going to be a bother now, but we're trying to have clean public API. This PR should make no changes to `data_model`

You can bypass orphan rule if you define `PartialHash` trait so you can implement for structs in `iroha_data_model` like `CommittedBlock`

Reply:
Users will want to have access to this value because it's very important for debugging. And all code that depended on the behavior of the old hash is to use this one instead because it is a preservation. Partial hash = Old Hash. Therefore it should stay.